### PR TITLE
Typewriter and Mobile Menus

### PR DESCRIPTION
**Pull request by WilliamHCarter**
1. Created `Typewriter` component for letter-by-letter rendering
2. Tweaked `ResponseHandler` for better input vetting
3. Created `MobileMenu` component for accessing navbar features on mobile devices

### CODE REVIEW

File: client/tailwind.config.js [hunk 1]
```diff
@@ -24,6 +24,9 @@ export default {
         'laptop': '1024px',
         'tablet': '600px',
       },
+      keyframes: {
+        
+      }
```
Comment (amchugh) [inline client/tailwind.config.js]: Is this needed?

File: client/src/components/Typewriter.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  useEffect(() => {
+    if (messageIndex < conversation.length) {
+      if (charIndex < conversation[messageIndex].length) {
+        const timeoutId = setTimeout(() => {
+          const newMessage = currentMessage + conversation[messageIndex][charIndex];
+          setCurrentMessage(newMessage);
+          onMessageUpdate(newMessage);  
+          setCharIndex((prev) => prev + 1);
+        }, 50); // Adjust speed here
```
Comment (amchugh) [inline client/src/components/Typewriter.tsx:21]: So there's no logic here reading the `character_delay_ms` and `do_type_message` fields, right? That'll be a later problem?

File: client/src/components/ResponseHandler.tsx [hunk 3]
```diff
@@ -35,12 +35,13 @@ function ResponseHandler() {
...
-      let options = data.response?.options ?? [];
-      let system = data.response?.system ? "\n> " + data.response.system : null;
+      let options = (data.response?.options ?? []).map((option: any[]) => option.join(': '));
 
       console.log(data);
-      setConversation((prev) => [...prev, "\nUser: "+userInput+"\n", message, ...options, ...messages, system]);
+      setConversation((prev) => [...prev, "\nUser: "+userInput+"\n", message, ...options]);
+      //if (!data.response?.options){
+      //  handleUserInput("");
+      //}
```
Comment (amchugh) [inline client/src/components/ResponseHandler.tsx]: Remove dead code!

File: client/src/components/ResponseHandler.tsx [hunk 4]
```diff
@@ -35,12 +35,13 @@ function ResponseHandler() {
     if (response.ok) {
       const data = await response.json();
       let message = data.response?.message ?? "";
-      let messages = data.response?.messages ?? [];
-      let options = data.response?.options ?? [];
-      let system = data.response?.system ? "\n> " + data.response.system : null;
+      let options = (data.response?.options ?? []).map((option: any[]) => option.join(': '));
```
Comment (amchugh) [inline client/src/components/ResponseHandler.tsx:38]: Clever!

**Review by amchugh — APPROVED** [review]
Minor comments

File: client/src/components/ResponseHandler.tsx (hunk 4 above)
Comment (WilliamHCarter) [reply client/src/components/ResponseHandler.tsx:38]: 🙏

File: client/src/components/ResponseHandler.tsx (hunk 3 above)
Comment (WilliamHCarter) [reply client/src/components/ResponseHandler.tsx]: Yep, this is here as a TODO for the next PR but thats bad practice, should've just left a comment

File: client/src/components/Typewriter.tsx (hunk 2 above)
Comment (WilliamHCarter) [reply client/src/components/Typewriter.tsx:21]: yep, later problem

File: client/tailwind.config.js (hunk 1 above)
Comment (WilliamHCarter) [reply client/tailwind.config.js]: Nope, good catch!

File: client/src/components/ResponseHandler.tsx (hunk 3 above)
Comment (amchugh) [reply client/src/components/ResponseHandler.tsx]: Makes sense. Didn't have the contextual knowledge to realize that. That's more ok :)

File: client/src/components/Typewriter.tsx (hunk 2 above)
Comment (amchugh) [reply client/src/components/Typewriter.tsx:21]: Sounds good

**Pull request merged**